solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -285,27 +316,124 @@ public boolean checkConfigExists(String configName) throws IOException {
   public abstract void copyConfig(String fromConfig, String toConfig, Set<String> copiedToZkPaths) throws IOException;
 
   /**
-   * Upload files from a given path to a config Zookeeper
+   * Create file path in config
    *
-   * @param dir        {@link java.nio.file.Path} to the files
-   * @param configName the name to give the config
-   * @throws IOException if an I/O error occurs or the path does not exist
+   * @param configName the config to download
+   * @param fileName  file path to be created
+   * @param failOnExists if file path already exists then do not re-create
+   * @throws IOException if an I/O error occurs
    */
-  public abstract void uploadConfig(Path dir, String configName) throws IOException;
+  public abstract void createFilePathInConfig(String configName, String fileName, boolean failOnExists) throws IOException;
```

Comment:
I'm skeptical this is necessary.  Instead, files can be uploaded that have slashes in there and we can create recursively as needed.